### ISSUE TITLE
Incorrectly parsed ss link

### ISSUE THREAD

**Issue by betaxab** [hiddify/hiddify-next#183]
### Search first

- [X] I searched and no similar issues were found

### What Happened?

Unable to parse chacha20-ietf-poly1305 protocol correctly, for example,

ss://Y2hhY2hhMjAtaWV0Zi1wb2x5MTMwNToxMWIyNDFjNi1jNzAxLTRkOGMtOGJhOC0zYzgzYmFkN2ViYmU@example.com:5000#Server1

base64 decode:
chacha20-ietf-poly1305:11b241c6-c701-4d8c-8ba8-3c83bad7ebbe

In the current-config.json, it will become:

{
      "type": "shadowsocks",
      "tag": "Server1 § 0",
      "server": "example.com",
      "server_port": 5000,
      "method": "Y2hhY2hhMjAtaWV0Zi1wb2x5MTMwNToxMWIyNDFjNi1jNzAxLTRkOGMtOGJhOC0zYzgzYmFkN2ViYmU",
      "password": ""
    },

### Reproduce the Bug

Add from Clipboard

### Version

v0.10.10 dev

### Platform/OS

Windows

### Are you willing to submit a PR? If you know how to fix the bug.

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by hiddify-com** [comment]
your code has error
There is a missing = at the end
```
ss://[Y2hhY2hhMjAtaWV0Zi1wb2x5MTMwNToxMWIyNDFjNi1jNzAxLTRkOGMtOGJhOC0zYzgzYmFkN2ViYmU=@example.com](mailto:Y2hhY2hhMjAtaWV0Zi1wb2x5MTMwNToxMWIyNDFjNi1jNzAxLTRkOGMtOGJhOC0zYzgzYmFkN2ViYmU@example.com):5000#Server1
```

**Comment by betaxab** [comment]
@hiddify-com Please look at the documentation:

https://shadowsocks.org/doc/sip002.html

According to standard SIP002, Base64URL encoding should be used instead of standard Base64.
Base64URL will be replaced as follows:

```
'+' => '-'
'/' => '_'
'=' => ''
```

About why should be use base64url, Please see the explanation:

https://base64.guru/standards/base64url


> The first problem is that the main standard uses “+” as the 62rd character and “=” as padding character. Both characters have a special meaning in the URI address: “+” is interpreted as space, while “=” is used to send data via query string as “key=value” pair. As you understand, using these symbols may lead to various errors.

**Comment by hiddify-com** [comment]
uh thanks i will take it into consideration

**Comment by hiddify-com** [comment]
I believe that this issue has been resolved in 0.12.0
